### DISCUSSION TITLE
Gene names

### DISCUSSION THREAD

**Discussion by lenawilde** [vdemichev/DiaNN#174] [Q&A]
Hi Vadim,

I am using a fasta file of a very rare organism, where the gene names are missing for many entries. I am doing Protein inference using "Genes", as it is by default and finally would like to use the "unique genes matrix" for further data evaluation. So my question is, what DIA-NN will do when it does not find gene names?

Many thanks in advance!

Best,
Lena

**Comment by vdemichev** [comment]
Hi Lena,

If gene names are missing for a substantial number of proteins, it's probably better to switch protein inference to "isoforms".
Can also try --relaxed-prot-inf, if too many isoforms per protein group complicate downstream workflows

Best,
Vadim

**Reply by lenawilde** [reply]
Hi Vadim,

you are amazing, ansewering so fast! Thank you so much!
I will try the --relaxed-prot-inf approach.

Many thanks,
Lena

**Reply by vdemichev** [reply]
I mean --relaxed-prot-inf in addition to switching to isoforms. "Isoforms" option is essential here, as proteins without gene names are simply ignored, if inference is set to "Genes"

**Reply by lenawilde** [reply]
Hi Vadim,

thanks for clarifying this. I will definitely change to "isoforms" and try relaxed inference in addition.

Can I ask you two more questions, just to be sure :-).
Because I have a two step process, first building libraries with GPF and than using them for my DIA analysis. So I think I should apply you suggestion in both steps, right?
Also, if I perform inference with "isoforms" and --relaxed-prot-inf, how will this change the unique genes matrix? Or do you suggest to stay with the protein gruops matrix for this case? (Just to clarify, the idea of my project is more or less benchmarking IDs from a very understudied species :-))

Best,
Lena


Best,

**Reply by vdemichev** [reply]
- No, I would disable protein grouping during the analysis step. I alsp recommend trying pure library-free analysis, might work better. Also, please see https://github.com/vdemichev/DiaNN/discussions/144.
- Unique genes matrix will still be there. But proteins which don't have gene names (regardless of the method used) will not be included there. So I suggest using the protein group matrix.

For benchmarking IDs: --relaxed-prot-inf is strongly recommended, and yes, better to use pg_matrix. Please also see the comments in the docs about run-specific and global protein FDR.

**Reply by vdemichev** [reply]
Hi Lena, please see below,the recommendation above to disable protein inference during analysis step is incorrect. Need to keep it on.

**Reply by lenawilde** [reply]
Hi Vadim,
many thanks for pointing this out. I will keep inferences on and as we counterchecked the amount of entries in the fasta file with missing gene names and the number ist not substantial we decided to stay with "genes".  

Thanks again for helping me, it is so much appreciated!
Lena

**Comment by bshashikadze** [comment]
Hi Vadim,

I am using similar approach to generate spectral library from GPF runs and use generated spectral library to analyse actual DIA runs. Actually, I am not sure about inference step during analysis-before I was actually leaving it to Genes in both step (lib generation and analysis), but from your reply also from manual it seems one can (or has to?) turn off inference in analysis step so DIA-NN will rely on grouping information from library (which makes sense since I had inference turned on during library generation step to Genes. Please note that I am not using relaxed protein inference, also I have not used “reannotate” function in any step, I somehow thought it was not necessary in this scenario).
Now, even though I have not noticed changes in main output (which I typically use for further analysis), I observed changes in pg matrix where I am confused. [using always the same spectral library] If I have inference off during analysis step pg matrix contains 15 050 entries, which is somehow a lot (for comparison same matrix from GPF fractions step contains 8975 entries), but if I leave inference to Genes during analysis then pg contains 6540 entries, I am not sure why?
Also if it is necessary to maintain fasta file in analysis step, because if I delete it in this step, it prints warning (“WARNING: 64575 precursors were wrongly annotated in the library as proteotypic”) regardless I have inference off or to Genes (so my guess is FASTA file should be present in library generation and analysis step as well?). So once again this changes nothing in main output (for example I have same number of proteotypic peptides), but why there is so big change in pg matrix when inference is off in analysis step? 
Thank you so much for clarification,
Bachi

**Reply by bshashikadze** [reply]
...Sorry, I said it changes nothing in main output but thats not true, even though number of unique peptides are always the same q-values are different in each case, as well as protein group column (which makes sence), but I am still not sure about big difference in pg when i have inference off in analysis step vs when it is set to Genes  :)

**Reply by vdemichev** [reply]
Hi Bachi, 

First, thanks to pointing this out. It was my mistake above, what I suggested does not make sense. DIA-NN always saves all proteins matching the peptide in the UniprotID column in the library, which is then used when the library is loaded. The ProteinGroup column in the library which actually contains inferred proteins is not used by DIA-NN itself. So to achieve what I suggested then one would need to replace the values in the UniprotID column with those from the ProteinGroup column. I guess I need to implement an option in DIA-NN to do this automatically.

So not to bother with this, I would suggest to just keep protein inference on when using DIA-NN-created libraries.

Best,
Vadim

**Reply by vdemichev** [reply]
Yes, FASTA needs to be included during the analysis step, if Protein inference is enabled

**Reply by bshashikadze** [reply]
Hi Vadim,
Great! thank you so much for the clarification! I will follow to your suggestion and keep inference on in analysis step then.
Just curiosity about inference to isoform IDs, is this working for swiss prot entries only? because, I mean, if I am not mistaken isoform IDs are present for curated entries only in Uniprot FASTA files am I right? so if one puts whole FASTA (curated + non-curated) and sets inference to isoform IDs then unique peptides will be counted taken into account swiss prot entries only?

Sorry for many questions and many thanks again, your help is appreciated a lot
Have a nice day,
Bachi
